src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -116,8 +116,12 @@ FilePath sessionLockFilePath(const FilePath& sessionDir)
    return sessionDir.complete("lock_file");
 }
 
-// session dir lock (initialized by attachToSourceDatabase)
-FileLock s_sessionDirLock;
+// session dir lock (lock is acquired within 'attachToSourceDatabase()')
+boost::shared_ptr<FileLock> createSessionDirLock()
+{
+   return FileLock::createDefault();
+}
+boost::shared_ptr<FileLock> s_pSessionDirLock = createSessionDirLock();
```

Comment:
I wonder if this should be a singleton-returning method rather than a static object, since it's not a POD type?